executor/builder.go

```diff
@@ -623,17 +623,30 @@ func (b *executorBuilder) buildSort(v *plan.Sort) Executor {
 }
 
 func (b *executorBuilder) buildNestedLoopJoin(v *plan.PhysicalHashJoin) *NestedLoopJoinExec {
-	bigExec := b.build(v.Children()[0])
-	smallExec := b.build(v.Children()[1])
+	if v.SmallTable == 1 {
+		return &NestedLoopJoinExec{
+			SmallExec:     b.build(v.Children()[1]),
+			BigExec:       b.build(v.Children()[0]),
+			Ctx:           b.ctx,
+			BigFilter:     expression.ComposeCNFCondition(b.ctx, v.LeftConditions...),
+			SmallFilter:   expression.ComposeCNFCondition(b.ctx, v.RightConditions...),
+			OtherFilter:   expression.ComposeCNFCondition(b.ctx, append(expression.ScalarFuncs2Exprs(v.EqualConditions), v.OtherConditions...)...),
```

Comment:
I found a maybe bug here but not yet verify it:
When v.EqualConditions is prepared, its column index is based on both left / right child schema. But when used in otherFilter, it needs to be based on a merged schema. Column Index in scalar args need a re-resolve.